src/form/form-fields-action-creator.ts

```diff
@@ -0,0 +1,30 @@
+import { createAction, createErrorAction } from '@bigcommerce/data-store';
+import { Observable, Observer } from 'rxjs';
+
+import { cachableAction, ActionOptions } from '../common/data-store';
+import { RequestOptions } from '../common/http-request';
+
+import { FormFieldsRequestSender } from '.';
+import { FormFieldsActionType, LoadFormFieldsAction } from './form-fields-actions';
+
+export default class FormFieldsActionCreator {
+    constructor(
+        private _formFieldsRequestSender: FormFieldsRequestSender
+    ) {}
+
+    @cachableAction
+    loadFormFields(options?: RequestOptions & ActionOptions): Observable<LoadFormFieldsAction> {
+        return Observable.create((observer: Observer<LoadFormFieldsAction>) => {
+            observer.next(createAction(FormFieldsActionType.LoadFormFieldsRequested));
+
+            this._formFieldsRequestSender.loadFields(options)
+                .then(response => {
```

Comment:
Prefer `async/await` for new code

Reply:
For some reason async/await doesn't seem to work in this scenario, not sure why but this is what I got:
```js
    loadFormFields(options?: RequestOptions & ActionOptions): Observable<LoadFormFieldsAction> {
        return Observable.create(async (observer: Observer<LoadFormFieldsAction>) => {
            observer.next(createAction(FormFieldsActionType.LoadFormFieldsRequested));

            try {
                const response = await this._formFieldsRequestSender.loadFields(options);

                observer.next(createAction(FormFieldsActionType.LoadFormFieldsSucceeded, response.body));
                observer.complete();
            } catch (error) {
                observer.error(createErrorAction(FormFieldsActionType.LoadFormFieldsFailed, error));
            }
        });
    }
```

With that code, checkout-action-creator.spec.ts fails because `LoadFormFieldsSucceeded` action is not present.

Reply:
I can't immediately tell what's wrong with the code at a glance. Perhaps use [SubscriptionsActionCreator](https://github.com/bigcommerce/checkout-sdk-js/blob/master/src/subscription/subscriptions-action-creator.ts) as an example for reference?